spec/topological_inventory/openshift/operations/core/authentication_retriever_spec.rb

```diff
@@ -5,9 +5,52 @@ module Openshift
     module Operations
       module Core
         RSpec.describe AuthenticationRetriever do
-          let(:subject) { described_class.new(endpoint.id) }
+          let(:subject) { described_class.new(123) }
+
+          let(:headers) { {"Content-Type" => "application/json"} }
+          let(:endpoint_authentications_url) do
+            "http://localhost:3000/api/topological-inventory/v0.0/endpoints/123/authentications"
+          end
+          let(:internal_authentications_url) do
+            "http://localhost:3000/api/internal/v0.0/authentications/321?expose_encrypted_attribute[]=password"
+          end
+          let(:endpoints_authentications_response) do
+            {
+              "data" => [{
+                "id" => 321
+              }]
+            }
+          end
+          let(:internal_authentication_response) { {:password => "token"} }
```

Comment:
The key will be a string, and the hash will include the rest of the regular attributes